tool/src/kotlin/mod.rs

```diff
@@ -974,11 +981,14 @@ returnVal.option() ?: return null
             closeable: bool,
         }
         let (slice_method, closeable): (Cow<'cx, str>, bool) = match slice_type {
-            Slice::Str(_, StringEncoding::UnvalidatedUtf16) => ("readUtf16".into(), true),
-            Slice::Str(_, _) => ("readUtf8".into(), true),
-            Slice::Primitive(_, _) => ("native".into(), true),
-            Slice::Strs(StringEncoding::UnvalidatedUtf16) => ("readUtf16s".into(), true),
-            Slice::Strs(_) => ("readUtf8s".into(), true),
+            Slice::Str(Some(_), StringEncoding::UnvalidatedUtf16) => ("borrowUtf16".into(), true),
+            Slice::Str(None, StringEncoding::UnvalidatedUtf16) => ("moveUtf16".into(), true),
+            Slice::Str(Some(_), _) => ("borrowUtf8".into(), true),
+            Slice::Str(None, _) => ("moveUtf8".into(), true),
+            Slice::Primitive(Some(_), _) => ("borrow".into(), true),
+            Slice::Primitive(_, _) => ("move".into(), true),
```

Comment:
question: shouldn't `move`s be not-closeable? Am I missing something?

It's confusing since in the generated code these aren't being closed, but i'm not sure how that happens

Reply:
> If the method taking this memory as input returns with a lifetime parameter, then this memory object is linked in the edges of the ouput

Which code does this?

Reply:
@jcrist1 rephrasing my question:


https://github.com/rust-diplomat/diplomat/blob/main/feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/Float64Vec.kt#L83-L90

https://github.com/rust-diplomat/diplomat/blob/main/feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/Float64Vec.kt#L122

In the first code, we call `vMem.close()` once we're done. In the second code, we don't. Which code triggers that?

Reply:
Aha, I see

https://github.com/rust-diplomat/diplomat/blob/2b903255187976779798fc89df3fee7298641c80/tool/src/kotlin/mod.rs#L1009

There was already some code dealing with maybe-ownedness here.

I was surprised to not see any changes around closing in this PR, but it looks like that match already happened to correctly handle those cases.

Reply:
Love it when Past Me wasn't an idiot (although he did forget a null check apparently. Thanks @emarteca)